### PR TITLE
DEX 556, 558, 561 multiple simple additions to AssetGroupSighting and…

### PR DESCRIPTION
**Pull request by hwindsor**
… Sighting responses

## Pull Request Overview

- Added Asset_group guid to Asset Group Sighting responses (DEX 558)
- Added Sighting Guid if present to Asset Group Sighting responses (DEX 561)
- Added Creator to AssetGroupSighting and Sighting responses (DEX 556)

---

**REST API Updates [OPTIONAL]**
Updated responses with updates in bold. Again if anyone can show me how to make github format this correctly as opposed to removing all spaces I'd be keen to hear
Asset Group Sighting: 
{
    **"asset_group_guid": "e90ae9fa-e70a-4417-a8e9-bf563780e3a9",**
    "assets": [
        {
            "filename": "zebra.jpg",
            "guid": "cb78099b-ded0-422c-a07d-9bb8900deef9",
            "src": "/api/v1/assets/src/cb78099b-ded0-422c-a07d-9bb8900deef9"
        }
    ],
    "completion": 10,
    "config": {
        "assetReferences": [
            "zebra.jpg"
        ],
        "encounters": [
            {
                "guid": "77b5641e-bf6d-49ed-8abb-4928defcfaab"
            }
        ],
        "locationId": "Tiddleywink",
        "startTime": "2000-01-01T01:01:01Z"
    },
    **"creator": {
        "full_name": "Public User",
        "guid": "0ad7a00b-c9d6-4a19-b14f-f12cacae7c8d",
        "profile_fileupload": null
    },**
    "guid": "08940b2a-1b3e-4167-82d8-168b1649964f",
    "jobs": {
        "03cfa564-7b6a-4397-8c5c-db74629c61d1": {
            "active": false,
            "asset_ids": [
                "cb78099b-ded0-422c-a07d-9bb8900deef9"
            ],
            "json_result": {
                "image_uuid_list": [
                    {
                        "__UUID__": "2d21ccc5-96e2-4785-b2d5-a2a024ac934e"
                    }
                ],
                "results_list": [
                    [
                        {
                            "class": "zebra_plains",
                            "confidence": 0.8568,
                            "height": 539,
                            "id": 1,
                            "interest": false,
                            "left": 459,
                            "multiple": false,
                            "quality": null,
                            "species": "zebra_plains",
                            "theta": 0.0,
                            "top": 126,
                            "uuid": {
                                "__UUID__": "1891ca05-5fa5-4e52-bb30-8ee80941c2fc"
                            },
                            "viewpoint": "test",
                            "width": 531,
                            "xtl": 459,
                            "ytl": 126
                        },
                        {
                            "class": "zebra_plains",
                            "confidence": 0.853,
                            "height": 500,
                            "id": 2,
                            "interest": false,
                            "left": 26,
                            "multiple": false,
                            "quality": null,
                            "species": "zebra_plains",
                            "theta": 0.0,
                            "top": 145,
                            "uuid": {
                                "__UUID__": "0c6f3a16-c3f0-4f8d-a47d-951e49b0dacb"
                            },
                           "viewpoint": "test",
                            "width": 471,
                            "xtl": 26,
                            "ytl": 145
                        }
                    ]
                ]
            },
            "model": "african_terrestrial",
            "start": "Tue, 23 Nov 2021 12:08:52 GMT"
        }
    },
    **"sighting_guid": null,**
    "stage": "curation"
}

Asset Group Sighting As Sighting: 
{
    **"asset_group_guid": "6b0d805d-bde1-4a00-bf83-dc9799912f06",**
    "assets": [
        {
            "annotations": [],
            "created": "2021-11-23T12:09:02.937765+00:00",
            "dimensions": {
                "height": 664,
                "width": 1000
            },
            "filename": "zebra.jpg",
            "guid": "05e6e0b7-bd96-4897-8150-6df681c6e376",
            "src": "/api/v1/assets/src/05e6e0b7-bd96-4897-8150-6df681c6e376",
            "updated": "2021-11-23T12:09:02.937784+00:00"
        }
    ],
    "comments": "None",
    "completion": 10,
    "createdEDM": null,
    "createdHouston": "2021-11-23T12:09:02.982969+00:00",
    **"creator": {
        "full_name": "First Middle Last",
        "guid": "680ecb05-c289-45ca-a4ab-90bcd78be826",
        "profile_fileupload": null
    },**
    "customFields": {},
    "decimalLatitude": null,
    "decimalLongitude": null,
    "encounterCounts": {},
    "encounters": [
        {
            "createdHouston": "2021-11-23T12:09:02.982969+00:00",
            "customFields": {},
            "decimalLatitude": null,
            "decimalLongitude": null,
            "guid": "7ee8d9ae-2dba-434e-926f-d69ed3e888d6",
            "hasEdit": true,
            "hasView": true,
            "id": "7ee8d9ae-2dba-434e-926f-d69ed3e888d6",
            "individual": {},
            "locationId": null,
            "owner": {
                "full_name": "First Middle Last",
                "guid": "18621e34-4d01-4eb2-84a4-0d89c75fb644",
                "profile_fileupload": null
            },
            "sex": null,
            "submitter": null,
            "taxonomy": {},
            "time": null,
            "updatedHouston": "2021-11-23T12:09:02.982980+00:00",
            "version": null
        }
    ],
    "featuredAssetGuid": null,
    "guid": "06515317-7349-40bc-98e2-5f20d7ef6bbd",
    "hasEdit": true,
    "hasView": true,
    "id": "06515317-7349-40bc-98e2-5f20d7ef6bbd",
    "locationId": "Tiddleywink",
    **"sightingGuid": null,**
    "stage": "curation",
    "startTime": "2000-01-01T01:01:01Z",
    "updatedHouston": "2021-11-23T12:09:02.982980+00:00",
    "verbatimLocality": "",
    "version": null
}

Sighting response

{
    "assets": [],
    "comments": "None",
    "createdEDM": "2021-11-23 12:13:49",
    "createdHouston": "2021-11-23T12:13:49.108220+00:00",
    **"creator": {
        "full_name": "First Middle Last",
        "guid": "7110b56c-f3bf-4fc9-a472-5dba1be15764",
        "profile_fileupload": null
    },**
    "customFields": {
        "47f3f896-dabf-43a1-a9be-40af79d5146f": "CFD_TEST_VALUE"
    },
    "decimalLatitude": 87.14680786933428,
    "decimalLongitude": -65.24900073490588,
    "encounterCounts": {
        "individuals": 0,
        "sex": {
            "male": 1
        }
    },
    "encounters": [
        {
            "createdHouston": "2021-11-23T12:13:49.112726+00:00",
            "customFields": {
                "f5538724-acd2-4953-b8cf-9ae27a2b638a": "CFD_TEST_VALUE"
            },
            "decimalLatitude": -66.94874070051998,
            "decimalLongitude": 23.151173260727433,
            "guid": "db5b5a29-f585-4d78-9f7f-677f8e342e0b",
            "hasEdit": true,
            "hasView": true,
            "id": "db5b5a29-f585-4d78-9f7f-677f8e342e0b",
            "individual": {},
            "locationId": "enc-test",
            "owner": {
                "full_name": "First Middle Last",
                "guid": "7110b56c-f3bf-4fc9-a472-5dba1be15764",
                "profile_fileupload": null
            },
            "sex": "male",
            "submitter": {
                "full_name": "First Middle Last",
                "guid": "7110b56c-f3bf-4fc9-a472-5dba1be15764",
                "profile_fileupload": null
            },
            "taxonomy": {
                "commonNames": [
                    "Example"
                ],
                "id": "f57320cf-7f98-4a14-93f2-81eeb3e0d29c",
                "scientificName": "Exempli gratia"
            },
            "time": "2021-11-23T12:13:49.067Z",
            "updatedHouston": "2021-11-23T12:13:49.112737+00:00",
            "version": 1637669629085
        }
    ],
    "endTime": "2021-11-23T13:13:49.066Z",
    "featuredAssetGuid": null,
    "guid": "5ebd2328-541c-4e56-8589-11663bfc39d6",
    "hasEdit": true,
    "hasView": true,
    "id": "5ebd2328-541c-4e56-8589-11663bfc39d6",
    "locationId": "sig-test",
    "stage": "processed",
    "startTime": "2021-11-23T12:13:49.066Z",
    "taxonomies": [
        {
            "commonNames": [
                "Example"
            ],
            "id": "f57320cf-7f98-4a14-93f2-81eeb3e0d29c",
            "scientificName": "Exempli gratia"
        }
    ],
    "updatedHouston": "2021-11-23T12:13:49.108238+00:00",
    "version": 1637669629087
}

### CODE REVIEW

File: app/modules/sightings/models.py [hunk 1]
```diff
@@ -101,6 +101,12 @@ def get_owner(self):
             return self.get_owners()[0]
         return None
 
+    def get_creator(self):
+        if self.asset_group_sighting:
```
Comment (hwindsor) [inline app/modules/sightings/models.py:105]: @jon, this if was intentionally to make your migration easier

File: tests/modules/asset_groups/resources/utils.py [hunk 2]
```diff
@@ -704,29 +704,56 @@ def patch_asset_group_sighting_as_sighting(
...
+            'stage',
+            'config',
+            'completion',
+            'assets',
+            'creator',
+            'asset_group_guid',
+            'sighting_guid',
+        },
     )
+    # so we can capture the example if we want, via -s flag on pytest
```
Comment (hwindsor) [inline tests/modules/asset_groups/resources/utils.py:724]: Stole this idea from one of the sighting tests. It's useful to have them always printed out with the -s flag but do you think this is a parrern we should be following for every read util in all the modules?

File: tests/modules/sightings/resources/utils.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def read_sighting(flask_app_client, user, sight_guid, expected_status_code=200):
         'sightings:read',
         f'{PATH}{sight_guid}',
         expected_status_code,
-        response_200={'id'},
+        response_200={'id', 'creator', 'encounters', 'stage', 'assets'},
```
Comment (hwindsor) [inline tests/modules/sightings/resources/utils.py:81]: Only a limited set of the fields. Do you think a fuller set should be here?

Comment (colinwkingen) [reply tests/modules/sightings/resources/utils.py:81]: This is fine for now. Not a big deal to add more if they are desired later.

File: tests/modules/asset_groups/resources/utils.py (hunk 2 above)
Comment (colinwkingen) [reply tests/modules/asset_groups/resources/utils.py:724]: Probably a useful thing to do as API's stop changing so quickly.

**Review by colinwkingen — APPROVED** [review]
LGTM

**Pull request merged**